### ISSUE TITLE
train error

### ISSUE THREAD

**Issue by zhuxiaoliu00** [ICSResearch/MTC-CSNet#1]
![image](https://github.com/ICSResearch/MTC-CSNet/assets/127464603/ccbcca98-d4f3-482e-a4f4-e3915cdde812)
I downloaded the source code directly for debugging, but I always encountered this error. I would like to ask what caused it.

**Comment by Zhihx** [comment]
> ![image](https://private-user-images.githubusercontent.com/127464603/323800132-ccbcca98-d4f3-482e-a4f4-e3915cdde812.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4OTA3OTMsIm5iZiI6MTcxOTg5MDQ5MywicGF0aCI6Ii8xMjc0NjQ2MDMvMzIzODAwMTMyLWNjYmNjYTk4LWQ0ZjMtNDgyZS1hNGY0LWUzOTE1Y2RkZTgxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQwMzIxMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MDBkN2VkZGUzNjdjZWU2YjBjMzJiNjIyYTgyY2QwZWZjMjgwMjc0MWVjNTg5NzYzZjUxZTU1MGMzMDI3Y2RjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7S-1I5Od2ofyaVWRVQ7clTkbZkRoV4bRpmO116shGr4) I downloaded the source code directly for debugging, but I always encountered this error. I would like to ask what caused it.

Hi, I am also testing this repository and also encountered this error. I found it is because of mismatched matrix dimension of ```x_temp``` and ```recon```. The deeper reason is a bug in ```patch2block``` function in demo.py. Although I find all these bugs in source code and attempts to train under BSDS500 dataset, I found the batch loss is quite large. So maybe I will try to figure out what reason is. Actually I still worry the source code cannot be feasiable for reproduction. Hope this is useful for your work.

**Comment by zhuxiaoliu00** [comment]
> > ![image](https://private-user-images.githubusercontent.com/127464603/323800132-ccbcca98-d4f3-482e-a4f4-e3915cdde812.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4OTA3OTMsIm5iZiI6MTcxOTg5MDQ5MywicGF0aCI6Ii8xMjc0NjQ2MDMvMzIzODAwMTMyLWNjYmNjYTk4LWQ0ZjMtNDgyZS1hNGY0LWUzOTE1Y2RkZTgxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQwMzIxMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MDBkN2VkZGUzNjdjZWU2YjBjMzJiNjIyYTgyY2QwZWZjMjgwMjc0MWVjNTg5NzYzZjUxZTU1MGMzMDI3Y2RjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7S-1I5Od2ofyaVWRVQ7clTkbZkRoV4bRpmO116shGr4) I downloaded the source code directly for debugging, but I always encountered this error. I would like to ask what caused it.
> 
> Hi, I am also testing this repository and also encountered this error. I found it is because of mismatched matrix dimension of `x_temp` and `recon`. The deeper reason is a bug in `patch2block` function in demo.py. Although I find all these bugs in source code and attempts to train under BSDS500 dataset, I found the batch loss is quite large. So maybe I will try to figure out what reason is. Actually I still worry the source code cannot be feasiable for reproduction. Hope this is useful for your work.

Ok, I will look at this error and hope I can solve it as soon as possible. Are you also studying the paper on compressed sensing recently? We can discuss it together sometime！

**Comment by Zhihx** [comment]
> > > ![image](https://private-user-images.githubusercontent.com/127464603/323800132-ccbcca98-d4f3-482e-a4f4-e3915cdde812.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4OTA3OTMsIm5iZiI6MTcxOTg5MDQ5MywicGF0aCI6Ii8xMjc0NjQ2MDMvMzIzODAwMTMyLWNjYmNjYTk4LWQ0ZjMtNDgyZS1hNGY0LWUzOTE1Y2RkZTgxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQwMzIxMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MDBkN2VkZGUzNjdjZWU2YjBjMzJiNjIyYTgyY2QwZWZjMjgwMjc0MWVjNTg5NzYzZjUxZTU1MGMzMDI3Y2RjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7S-1I5Od2ofyaVWRVQ7clTkbZkRoV4bRpmO116shGr4) I downloaded the source code directly for debugging, but I always encountered this error. I would like to ask what caused it.
> > 
> > 
> > Hi, I am also testing this repository and also encountered this error. I found it is because of mismatched matrix dimension of `x_temp` and `recon`. The deeper reason is a bug in `patch2block` function in demo.py. Although I find all these bugs in source code and attempts to train under BSDS500 dataset, I found the batch loss is quite large. So maybe I will try to figure out what reason is. Actually I still worry the source code cannot be feasiable for reproduction. Hope this is useful for your work.
> 
> Ok, I will look at this error and hope I can solve it as soon as possible. Are you also studying the paper on compressed sensing recently? We can discuss it together sometime！

Actually I learn the paper by studying the source code step by step. Actually I found several errors in the source code and hope I can fix it. I am trying to train the network at learning rate of 0.1 right now. Hope I can get a good result.
